### DISCUSSION TITLE
Delphi Horse - Character "\" on response

### DISCUSSION THREAD

**Discussion by vianaprod** [HashLoad/horse#367] [Q&A]
Estou tentando resolver esse problema há algum tempo, mas não consigo.
Sempre retorna: { "Numero": "AFR\/DEF-0007899" }

![teste1_11-12-2023](https://github.com/HashLoad/horse/assets/25791890/335da289-a542-48af-8ef4-78d68c3aa82f)

Code:

uses
  Horse;

begin
  THorse.Get('/teste1',
    procedure(Req: THorseRequest; Res: THorseResponse)
    var
      JSONData: string;
    begin
      JSONData := '{"Numero": "AFR/DEF-0007899"}';

      Res.ContentType('application/json');

      Res.Send(JSONData);
    end);

  THorse.Listen(9000);
end.

**Comment by alcantarajoaopaulo** [comment]
Olá

Na verdade não entendi o problema. O retorno está correto, conforme alimentação manual da variável.

**Reply by vianaprod** [reply]
Obrigado pelo retorno!
Esse foi um teste usando a string, mas tenho uma tabela no BD que tem um campo exatamente com esse formato.
Mesmo usando FDQuery.ToJsonObject retorna com essa `\` a mais.
Agradeço muito se puder me ajudar com algum exemplo ou uma ideia.

**Comment by alcantarajoaopaulo** [comment]
Ah sim, entendido.
Não se preocupe com elas; isso não é um problema.

A barra é adicionada automaticamente por ser um caractere de escape.
Como /n , por exemplo, é um comando de quebra de linha,
caracteres de escape como a barra invertida informa ao interpretador um sinal do tipo "_A barra a seguir não é um comando, e sim o número do pedido_"
Normal.

O Postman mostra ela, mas o seu front-end ao colocar esse conteúdo no campo, não irá capturar a barra.

Edit: E isso não é só à nível Horse, à nível de Delphi. É uma convenção nativa do próprio JSON.

**Reply by vianaprod** [reply]
Show de bola, agora entendi!
Esse foi o primeiro caso que peguei com caráter especial.
Muito Obrigado pela ajuda!!
